src/api/tmc.ts

```diff
@@ -258,11 +304,97 @@ export default class TMC {
+        courseSlug: string,
+        exerciseChecksum: string,
+        exerciseId: number,
+        exercisePath: string,
+        exerciseSlug: string,
+    ): Promise<Result<void, Error>> {
+        const onStdout = (res: LangsStatusUpdate<unknown>): void => {
+            Logger.debug(res);
+        };
+
+        return this._executeLangsCommand(
+            {
+                args: [
+                    "settings",
+                    "--client-name",
+                    this.clientName,
+                    "migrate",
+                    "--course-slug",
+                    courseSlug,
+                    "--exercise-checksum",
+                    exerciseChecksum,
+                    "--exercise-id",
+                    `${exerciseId}`,
+                    "--exercise-path",
+                    exercisePath,
+                    "--exercise-slug",
+                    exerciseSlug,
+                ],
+                core: false,
+                onStdout,
+            },
+            createIs<unknown>(),
+        ).then((res) => (res.err ? res : Ok.EMPTY));
+    }
+
+    public async getSetting(key: string): Promise<Result<string, Error>> {
+        return (await this.getSettingObject(key, createIs<string>())).map((x) => x ?? "");
+    }
+
+    public async getSettingObject<T>(
+        key: string,
+        checker: (object: unknown) => object is T,
+    ): Promise<Result<T | undefined, Error>> {
+        return this._executeLangsCommand(
+            {
+                args: ["settings", "--client-name", this.clientName, "get", key],
+                core: false,
+            },
+            checker,
+        ).then((res) => res.map((x) => x.data ?? undefined));
+    }
+
+    public async setSetting(key: string, value: string): Promise<Result<void, Error>> {
+        return this._executeLangsCommand(
+            {
+                args: ["settings", "--client-name", this.clientName, "set", key, value],
+                core: false,
+            },
+            createIs<unknown>(),
+        ).then((x) => x.andThen(() => Ok.EMPTY));
+    }
+
     // ---------------------------------------------------------------------------------------------
     // Core commands
     // ---------------------------------------------------------------------------------------------
 
     /**
+     * Checks for updates for all exercises in this client's context. Uses TMC-langs
+     * `check-exercise-updates` core command internally.
+     */
+    public async checkExerciseUpdates(
+        options?: CacheOptions,
+    ): Promise<Result<Array<{ id: number }>, Error>> {
+        return this._executeLangsCommand(
+            { args: ["check-exercise-updates"], core: true },
+            createIs<Array<{ id: number }>>(),
+            { forceRefresh: options?.forceRefresh, key: TMC._exerciseUpdatesCacheKey },
+        ).then((res) => (res.err ? res : Ok(res.val.data)));
+    }
+
+    /**
+     * @deprecated - Migrate to `downloadExercises`
```

Comment:
Should this be removed then?

Reply:
This function is still used in some tests, e.g. downloading an old submission requires an exercise to be present in the first place. This requires larger changes as the new `downloadExercises` uses external (langs) configuration to determine where downloads should go.